packages/yoroi-extension/app/UI/features/swap-new/common/components/Modals/DisclaimerDialog.tsx

```diff
@@ -0,0 +1,135 @@
+                disclaimerAgreed={disclaimerAgreed}
+                setDisclaimerAgreed={setDisclaimerAgreed}
+                action={action}
+              />
+            ),
+            height: '588px',
+            width: '702px',
+            modalId: 'swapDisclaimer',
+            onClose: () => {
+              navigate.walletTransactions();
+              localStorage.setSwapDisclaimerModalClosed('false');
+            },
+          });
+        }
+      } catch (error) {
+        console.error('Error checking modal state:', error);
+      }
+    };
+
+    checkModalState();
+  }, [disclaimerAgreed]);
+
+  return <></>;
+};
+
+const DisclaimerDialogBody = ({ disclaimerAgreed, setDisclaimerAgreed, action }) => {
+  const strings = useStrings();
+
+  return (
+    <Stack>
+      <Box display="flex" maxWidth="648px" flexDirection="column" gap="24px">
+        <Box>
+          <Typography component="div" variant="body1" color="grayscale.900" align="justify">
+            {strings.disclaimerDescription}
+          </Typography>
+        </Box>
+        <Box>
+          <Typography component="div" fontWeight={500} variant="body1" color="grayscale.900" align="justify">
+            {strings.disclaimerPleaseNote}
+          </Typography>
+          <Box
+            sx={{
+              display: 'flex',
+              alignItems: 'center',
+              flexFlow: 'column',
+            }}
+          >
+            {[strings.disclaimerNote1, strings.disclaimerNote2, strings.disclaimerNote3, strings.disclaimerNote4].map(
+              (message, i) => (
+                <Box
+                  sx={{
+                    display: 'flex',
+                    flexFlow: 'row nowrap',
+                    alignItems: 'flex-start',
+                    justifyContent: 'flex-start',
+                  }}
+                >
+                  <Typography component="div" variant="body1" color="grayscale.900">
+                    {i + 1}.&nbsp;
+                  </Typography>
+                  <Typography component="div" variant="body1" color="grayscale.900">
+                    {message}
+                  </Typography>
+                </Box>
+              )
+            )}
+          </Box>
+        </Box>
+
+        <FormControlLabel
+          label={
+            <Typography component="div" variant="body1" color="grayscale.900">
+              {strings.disclaimerCheckboxLabel}
+            </Typography>
+          }
+          control={
+            <Checkbox
+              onChange={() => setDisclaimerAgreed(!disclaimerAgreed)}
+              checked={disclaimerAgreed}
+              sx={{ marginRight: '8px' }}
+            />
+          }
+          sx={{ margin: '0px' }}
+        />
+      </Box>
+
+      <Stack pt={20}>
+        <LoadingButton
+          // @ts-ignore
+          variant="primary"
+          loading={action.isSubmitting}
+          onClick={action.onClick}
+          disabled={action.disabled === true}
```

Comment:
Since action.disabled is a boolean, no need to explicitly check if its true

```suggestion
          disabled={action.disabled }
```